.github/workflows/functional_mock_tests.yml

```
@@ -0,0 +1,75 @@
+name: Run functional mock tests
+on:
+  push:
+    branches:
+      - 'develop'
+      - 'master'
+  pull_request:
+    branches:
+      - 'master'
+      - 'develop'
+
+jobs:
+  tests_asan:
+    name: Run functional mock tests with AddressSanitizer
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4.1.7
+        with:
+          submodules: recursive
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.8'
+
+      - name: Install dependencies
+        run: |
+            sudo apt-get install cmake build-essential
+            pip install jsonschema jinja2
+
+      - name: Compile functional mock tests with AddressSanitizer
+        run: |
+            cd tests/functional_mock
+            mkdir -p build
+            cd build
+            cmake -DLT_ASAN=1 ..
+            make -j4
```

Let's use Ninja to have faster builds.
```suggestion
            cmake -DLT_ASAN=1 -G Ninja ..
            ninja
```